### PR TITLE
Throwing runtime fix

### LINKED ISSUE #17141: Runtime when throwing a mob at a human
@KorPhaeron #13987

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
@phil235 

Fixes #17141

### CODE REVIEW

File: code/modules/mob/living/carbon/human/human_defense.dm [hunk 1]
```diff
@@ -350,22 +350,22 @@
...
-					I.loc = src
-					L.take_damage(I.w_class*I.embedded_impact_pain_multiplier)
-					visible_message("<span class='danger'>\the [I.name] embeds itself in [src]'s [L.name]!</span>","<span class='userdanger'>\the [I.name] embeds itself in your [L.name]!</span>")
-					hitpush = 0
-					skipcatch = 1 //can't catch the now embedded item
+		if(I.thrownby != src && check_shields(throwpower, "\the [AM.name]", AM, THROWN_PROJECTILE_ATTACK))
+			hitpush = 0
+			skipcatch = 1
+			blocked = 1
+		else if(I)
```
Comment (phil235) [inline code/modules/mob/living/carbon/human/human_defense.dm]: no, we want this to be called when the istype(AM, /obj/item) above is true and set I = AM

File: code/modules/mob/living/carbon/human/human_defense.dm [hunk 2]
```diff
@@ -350,22 +350,22 @@
...
-					throw_alert("embeddedobject", /obj/screen/alert/embeddedobject)
-					var/obj/item/bodypart/L = pick(bodyparts)
-					L.embedded_objects |= I
-					I.add_blood(src)//it embedded itself in you, of course it's bloody!
-					I.loc = src
-					L.take_damage(I.w_class*I.embedded_impact_pain_multiplier)
-					visible_message("<span class='danger'>\the [I.name] embeds itself in [src]'s [L.name]!</span>","<span class='userdanger'>\the [I.name] embeds itself in your [L.name]!</span>")
-					hitpush = 0
-					skipcatch = 1 //can't catch the now embedded item
+		if(I.thrownby != src && check_shields(throwpower, "\the [AM.name]", AM, THROWN_PROJECTILE_ATTACK))
```
Comment (phil235) [inline code/modules/mob/living/carbon/human/human_defense.dm]: we want check_shield() to be called when the thrown thing is a mob

Comment (KorPhaeron) [inline code/modules/mob/living/carbon/human/human_defense.dm]: I thought we removed any negative effects from having another mob thrown at you?

Comment (optimumtact) [inline code/modules/mob/living/carbon/human/human_defense.dm]: Any resolution to this?

Comment (phil235) [inline code/modules/mob/living/carbon/human/human_defense.dm]: Kor needs to fix the issue he introduced in #13987 .
check_shields() needs to be called when the thrown thing is a mob, even if it currently does nothing on impact.

Comment (KorPhaeron) [inline code/modules/mob/living/carbon/human/human_defense.dm]: Fair enough

**Pull request closed without merging**